lib/chef/provider/git.rb

```diff
@@ -41,6 +41,20 @@ def load_current_resource
       end
 
       def define_resource_requirements
+        unless new_resource.user.nil?
+          requirements.assert(:all_actions) do |a|
+            a.assertion {
+              begin
+                require "etc" unless defined?(Etc)
+                get_dir(new_resource.user)
+              rescue ArgumentError
+                false
+              end
```

Comment:
There should probably be a `safe_get_homedir()` method extracted which rescues and returns `nil` (you can then `!!safe_get_homedir(new_resource.user)` here)